src/library/scala/collection/immutable/TreeSet.scala

```diff
@@ -40,6 +40,8 @@ final class TreeSet[A] private (tree: RB.Tree[A, Unit])(implicit val ordering: O
 
   override def size: Int = RB.count(tree)
 
+  override def isEmpty = size == 0
```

Comment:
`RB.isEmpty(tree)` is probably cheaper

Also, we'll most likely want to fixe the iterators in RedBlackTree since they allocate new iterators unconditionally: https://github.com/retronym/scala/blob/ec5d494534415fe8d4f2ecdec3bdb22c3a0d9487/src/library/scala/collection/immutable/RedBlackTree.scala#L138

Reply:
@retronym Not sure you saw these comments.